### ISSUE TITLE
Hanging on x86

### ISSUE THREAD

**Issue by divergentdave** [spacejam/sled#656]
Thanks for your work on sled! I've been excited to incorporate it into a hobby project of mine.

I have found that sled hangs on 32-bit architectures. I made a [small test program](https://github.com/divergentdave/sled_windows_minimized_test_case) to demonstrate the issue. This program reliably hangs and pegs the CPU while dropping the tree when compiled for i686, and it exits cleanly when compiled for x86_64, both on Linux and on Windows.

- I started out using sled 0.23.0, but also reproduced the issue with the latest revision, ff5fa3813ba6e9131ecfcac356b52618aa676c61.
- I am using rustc 1.34.1, on all platforms.
- I tested this on both Debian Stretch and Windows Server 2019 Datacenter.
- I compiled for the targets i686-unknown-linux-gnu and i686-pc-windows-msvc.

Please let me know if there's anything else you need, I'd be happy to help!

**Comment by spacejam** [comment]
@divergentdave thank you for the program to reproduce this! I'll hopefully have a fix shortly, and this is such a simple program that it clearly means I need to be adding an x86 test target for CI to catch this stuff ASAP

**Comment by divergentdave** [comment]
Sounds great! FYI I've been picking through the test suite on i686-unknown-linux-gnu, and dozens of tests hang. No test failures yet. There are hanging tests in test_pagecache.rs, so it's probably an issue in that layer and not sled.

**Comment by spacejam** [comment]
@divergentdave about to publish the fix, which can be applied by just doing "cargo update", bumping pagecache to `0.16.3`

**Comment by divergentdave** [comment]
I did an update, and I'm now using version 0.16.3. I get several of the following errors. Was there an issue with the release, or does sled need to update for the breaking API change too? I notice https://github.com/spacejam/sled/blob/master/crates/pagecache/Cargo.toml still shows version 0.16.2.

```
error[E0107]: wrong number of type arguments: expected 1, found 0
    --> /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/sled-0.23.0/src/tree.rs:1139:17
     |
1139 |         tx: &'g Tx,
     |                 ^^ expected 1 type argument
```

Crate hashes:
```
"checksum pagecache 0.16.3 (registry+https://github.com/rust-lang/crates.io-index)" = "48f795c392932b216169022935006c1af389b45a8769c269dc9615831772bea0"
"checksum sled 0.23.0 (registry+https://github.com/rust-lang/crates.io-index)" = "73a25c24d5f135b43b94e39f5722dbaf3923152e7fbc6dfa85cc800aeae09172"
```

**Comment by spacejam** [comment]
@divergentdave ahh duh I totally forgot about an API change, I just yanked that and published sled 0.24.1 which should be consistent again. sorry for that sloppy release, and thanks again for reporting this!
